server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/EmailSetMethodContract.scala

```diff
@@ -80,44 +68,222 @@ trait EmailSetMethodContract {
-           |      }, "c1"],
-           |      ["Email/get", {
-           |        "accountId": "29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6",
-           |        "state": "000001",
-           |        "list": [],
-           |        "notFound": ["${messageId1.serialize}"]
-           |      }, "c2"]
-           |    ]
-           |}""".stripMargin)
+    val response = `given`
+      .header(ACCEPT.toString, ACCEPT_RFC8621_VERSION_HEADER)
+      .body(request)
+    .when
+      .post
+    .`then`
+      .statusCode(SC_OK)
+      .contentType(JSON)
+      .extract
+      .body
+      .asString
+
+    assertThatJson(response).isEqualTo(
+      s"""{
+         |    "sessionState": "75128aab4b1b",
+         |    "methodResponses": [
+         |      ["Email/set", {
+         |        "accountId": "29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6",
+         |        "newState": "000001",
+         |        "destroyed": ["${messageId.serialize}"]
+         |      }, "c1"],
+         |      ["Email/get", {
+         |        "accountId": "29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6",
+         |        "state": "000001",
+         |        "list": [],
+         |        "notFound": ["${messageId.serialize}"]
+         |      }, "c2"]
+         |    ]
+         |}""".stripMargin)
+  }
+
+  @Test
+  def emailSetDestroyShouldFailWhenInvalidMessageId(server: GuiceJamesServer): Unit = {
+    val request =
+      s"""{
+         |  "using": [
+         |    "urn:ietf:params:jmap:core",
+         |    "urn:ietf:params:jmap:mail"],
+         |  "methodCalls": [
+         |    ["Email/set", {
+         |      "accountId": "29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6",
+         |      "destroy": ["invalid"]
+         |    }, "c1"]]
+         |}""".stripMargin
+
+    val response = `given`
+      .header(ACCEPT.toString, ACCEPT_RFC8621_VERSION_HEADER)
+      .body(request)
+    .when
+      .post
+    .`then`
+      .statusCode(SC_OK)
+      .contentType(JSON)
+      .extract
+      .body
+      .asString
+
+    assertThatJson(response).isEqualTo(
+      s"""{
+         |    "sessionState": "75128aab4b1b",
+         |    "methodResponses": [
+         |      ["error", {
+         |        "type": "invalidArguments",
+         |        "description": "{\\"errors\\":[{\\"path\\":\\"obj.destroy[0]\\",\\"messages\\":[\\"Invalid messageId\\"]}]}"
+         |      }, "c1"]
+         |    ]
+         |}""".stripMargin)
+  }
+
+  @Test
+  def emailSetDestroyShouldReturnNullWhenMessageIdNotFound(server: GuiceJamesServer): Unit = {
+    val request =
+      s"""{
+         |  "using": [
+         |    "urn:ietf:params:jmap:core",
+         |    "urn:ietf:params:jmap:mail"],
+         |  "methodCalls": [
+         |    ["Email/set", {
+         |      "accountId": "29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6",
+         |      "destroy": ["1"]
```

Comment:
This test will fail for distributed version.

You should do like for EmailSetMethodContract : let the test implementation generate a random messageId